### PR TITLE
feature: add antlr for mysql sqlparser

### PR DESCRIPTION
**Pull request by h-zhi**

### CODE REVIEW

File: sqlparser/seata-sqlparser-antlr/pom.xml [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <parent>
+        <groupId>io.seata</groupId>
+        <artifactId>seata-sqlparser</artifactId>
+        <version>${revision}</version>
+    </parent>
+    <modelVersion>4.0.0</modelVersion>
+    <artifactId>seata-sqlparser-antlr</artifactId>
+    <name>seata-sqlparser-druid ${project.version}</name>
```
Comment (slievrly) [inline sqlparser/seata-sqlparser-antlr/pom.xml:26]: antlr

**Comment by jsbxyyx** [comment]
has code not committed.

File: sqlparser/seata-sqlparser-antlr/src/main/java/io/seata/sqlparser/antlr/mysql/AntlrMySQLInsertRecognizer.java [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+    }
+
+    @Override
+    public String getTableName() {
+        return sqlContext.tableName;
+    }
+
+    @Override
+    public String getOriginalSQL() {
+        return null;
```
Comment (jsbxyyx) [inline sqlparser/seata-sqlparser-antlr/src/main/java/io/seata/sqlparser/antlr/mysql/AntlrMySQLInsertRecognizer.java:70]: return null?

File: sqlparser/seata-sqlparser-antlr/src/main/java/io/seata/sqlparser/antlr/mysql/AntlrMySQLInsertRecognizer.java [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+    }
+
+    @Override
+    public boolean insertColumnsIsEmpty() {
+        return false;
```
Comment (jsbxyyx) [inline sqlparser/seata-sqlparser-antlr/src/main/java/io/seata/sqlparser/antlr/mysql/AntlrMySQLInsertRecognizer.java:75]: return false?

File: sqlparser/seata-sqlparser-antlr/src/main/java/io/seata/sqlparser/antlr/mysql/AntlrMySQLInsertRecognizer.java [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+    }
+
+    @Override
+    public List<String> getInsertColumns() {
+        return null;
```
Comment (jsbxyyx) [inline sqlparser/seata-sqlparser-antlr/src/main/java/io/seata/sqlparser/antlr/mysql/AntlrMySQLInsertRecognizer.java:80]: return null?

File: sqlparser/seata-sqlparser-antlr/src/main/java/io/seata/sqlparser/antlr/mysql/AntlrMySQLInsertRecognizer.java [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+    }
+
+    @Override
+    public List<List<Object>> getInsertRows(Collection<Integer> primaryKeyIndex) {
+        return null;
```
Comment (jsbxyyx) [inline sqlparser/seata-sqlparser-antlr/src/main/java/io/seata/sqlparser/antlr/mysql/AntlrMySQLInsertRecognizer.java:85]: need implemention logic.

File: sqlparser/seata-sqlparser-antlr/src/main/java/io/seata/sqlparser/antlr/mysql/AntlrMySQLSelectRecognizer.java [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+    }
+
+    @Override
+    public String getTableName() {
+        return sqlContext.tableName;
+    }
+
+    @Override
+    public String getOriginalSQL() {
+        return null;
```
Comment (jsbxyyx) [inline sqlparser/seata-sqlparser-antlr/src/main/java/io/seata/sqlparser/antlr/mysql/AntlrMySQLSelectRecognizer.java:78]: return null?

File: sqlparser/seata-sqlparser-antlr/src/main/java/io/seata/sqlparser/antlr/mysql/AntlrMySQLSelectRecognizer.java [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+    }
+
+    @Override
+    public SQLType getSQLType() {
+        return SQLType.SELECT;
+    }
+
+    @Override
+    public String getTableAlias() {
+        return null;
```
Comment (jsbxyyx) [inline sqlparser/seata-sqlparser-antlr/src/main/java/io/seata/sqlparser/antlr/mysql/AntlrMySQLSelectRecognizer.java:68]: return null?

File: sqlparser/seata-sqlparser-antlr/src/main/java/io/seata/sqlparser/antlr/mysql/AntlrMySQLUpdateRecognizer.java [hunk 8]
```diff
@@ -0,0 +1,96 @@
...
+        for (MySqlContext.SQL sql : updateFoColumnNames) {
+            String.add(sql.getUpdateColumn());
+        }
+        return String;
+    }
+
+    @Override
+    public List<Object> getUpdateValues() {
+
+        return null;
```
Comment (jsbxyyx) [inline sqlparser/seata-sqlparser-antlr/src/main/java/io/seata/sqlparser/antlr/mysql/AntlrMySQLUpdateRecognizer.java:64]: return null?

File: sqlparser/seata-sqlparser-antlr/src/main/java/io/seata/sqlparser/antlr/mysql/AntlrMySQLUpdateRecognizer.java [hunk 9]
```diff
@@ -0,0 +1,96 @@
...
+    }
+
+    @Override
+    public String getWhereCondition(ParametersHolder parametersHolder, ArrayList<List<Object>> paramAppenderList) {
+        return null;
```
Comment (jsbxyyx) [inline sqlparser/seata-sqlparser-antlr/src/main/java/io/seata/sqlparser/antlr/mysql/AntlrMySQLUpdateRecognizer.java:69]: return null?

File: sqlparser/seata-sqlparser-antlr/src/main/java/io/seata/sqlparser/antlr/mysql/AntlrMySQLUpdateRecognizer.java [hunk 10]
```diff
@@ -0,0 +1,96 @@
...
+    }
+
+    @Override
+    public SQLType getSQLType() {
+        return SQLType.UPDATE;
+    }
+
+    @Override
+    public String getTableAlias() {
+        return null;
```
Comment (jsbxyyx) [inline sqlparser/seata-sqlparser-antlr/src/main/java/io/seata/sqlparser/antlr/mysql/AntlrMySQLUpdateRecognizer.java:84]: return null?

File: sqlparser/seata-sqlparser-antlr/src/main/java/io/seata/sqlparser/antlr/mysql/AntlrMySQLUpdateRecognizer.java [hunk 11]
```diff
@@ -0,0 +1,96 @@
...
+    }
+
+    @Override
+    public String getTableName() {
+        return sqlContext.tableName;
+    }
+
+    @Override
+    public String getOriginalSQL() {
+        return null;
```
Comment (jsbxyyx) [inline sqlparser/seata-sqlparser-antlr/src/main/java/io/seata/sqlparser/antlr/mysql/AntlrMySQLUpdateRecognizer.java:94]: return null?

File: sqlparser/seata-sqlparser-antlr/src/main/java/io/seata/sqlparser/antlr/mysql/AntlrMySQLDeleteRecognizer.java [hunk 12]
```diff
@@ -0,0 +1,80 @@
...
+    }
+
+    @Override
+    public String getTableName() {
+        return sqlContext.tableName;
+    }
+
+    @Override
+    public String getOriginalSQL() {
+        return null;
```
Comment (l81893521) [inline sqlparser/seata-sqlparser-antlr/src/main/java/io/seata/sqlparser/antlr/mysql/AntlrMySQLDeleteRecognizer.java:78]: Please return the statementProxy.getTargetSQL();

File: sqlparser/seata-sqlparser-antlr/src/main/java/io/seata/sqlparser/antlr/mysql/AntlrMySQLInsertRecognizer.java [hunk 13]
```diff
@@ -0,0 +1,100 @@
...
+    public String getOriginalSQL() {
+        return sqlContext.getOriginalSQL();
+    }
+
+    @Override
+    public boolean insertColumnsIsEmpty() {
+
+        List<MySqlContext.SQL> insertColumnNames = sqlContext.getInsertColumnNames();
+
+        if (insertColumnNames.isEmpty()) return true;
```
Comment (l81893521) [inline sqlparser/seata-sqlparser-antlr/src/main/java/io/seata/sqlparser/antlr/mysql/AntlrMySQLInsertRecognizer.java:81]: Use the style of
```
if() {

}
```

File: pom.xml [hunk 14]
```diff
@@ -269,7 +269,9 @@
                                     <failOnMissing>false</failOnMissing>
                                     <licenseMerges>
                                         <licenseMerge>Apache License, Version 2.0|The Apache Software License, Version
-                                            2.0|ASF 2.0|Apache 2|Apache-2.0|Apache 2.0 License|Apache 2.0|Apache License v2.0|Apache License 2.0|The Apache License, Version 2.0|The Apache Software License, Version 2.0
+                                            2.0|ASF 2.0|Apache 2|Apache-2.0|Apache 2.0 License|Apache 2.0|Apache License
```
Comment (l81893521) [inline pom.xml:272]: Please do not change the style.

File: pom.xml [hunk 15]
```diff
@@ -319,6 +319,9 @@
                             </includes>
                             <excludes>
                                 <exclude>**/generated/**</exclude>
+                                <exclude>**/antlr/mysql/parser/*.*</exclude>
+                                <exclude>**/antlr/mysql/antlr/*.*</exclude>
+                                <exclude>**/antlr/mysql/stream/ANTLRNoCaseStringStream.java</exclude>
```
Comment (l81893521) [inline pom.xml:324]: I think this class should not exclude.

File: bom/pom.xml [hunk 16]
```diff
@@ -97,6 +97,8 @@
         <sofa.registry.version>5.2.0</sofa.registry.version>
         <httpclient.version>4.5.8</httpclient.version>
         <httpcore.version>4.4.11</httpcore.version>
+        <druid.version>1.1.12</druid.version>
```
Comment (l81893521) [inline bom/pom.xml:100]: Please don't change the druid version.

File: pom.xml [hunk 17]
```diff
@@ -337,7 +340,8 @@
                                 <phase>validate</phase>
                                 <configuration>
                                     <configLocation>${user.dir}/style/seata_checkstyle.xml</configLocation>
-                                    <suppressionsLocation>${user.dir}/style/seata_suppressions.xml</suppressionsLocation>
+                                    <suppressionsLocation>${user.dir}/style/seata_suppressions.xml
```
Comment (l81893521) [inline pom.xml:343]: Please revert here.

File: pom.xml [hunk 18]
```diff
@@ -368,7 +372,8 @@
                 <artifactId>protobuf-maven-plugin</artifactId>
                 <version>${protobuf-maven-plugin.version}</version>
                 <configuration>
-                    <protoSourceRoot>${project.basedir}/src/main/resources/protobuf/io/seata/protocol/transcation/</protoSourceRoot>
+                    <protoSourceRoot>${project.basedir}/src/main/resources/protobuf/io/seata/protocol/transcation/
```
Comment (l81893521) [inline pom.xml:375]: Please reset here.

File: sqlparser/seata-sqlparser-antlr/src/main/java/io/seata/sqlparser/antlr/SQLOperateRecognizerHolderFactory.java [hunk 19]
```diff
@@ -0,0 +1,46 @@
...
+    private static final Map<String, SQLOperateRecognizerHolder> RECOGNIZER_HOLDER_MAP = new ConcurrentHashMap<>();
+
+    /**
+     * get SQLOperateRecognizer by db type
+     *
+     * @param dbType the db type
+     * @return the SQLOperateRecognizer
+     */
+    public static SQLOperateRecognizerHolder getSQLRecognizerHolder(String dbType) {
+        if (RECOGNIZER_HOLDER_MAP.get(dbType) != null) {
```
Comment (l81893521) [inline sqlparser/seata-sqlparser-antlr/src/main/java/io/seata/sqlparser/antlr/SQLOperateRecognizerHolderFactory.java:39]: EnhancedServiceLoader  already handle singleton, you can check SQLOperateRecognizerHolderFactory in druild.

File: bom/pom.xml (hunk 16 above)
Comment (h-zhi) [reply bom/pom.xml]: The druid version has been rolled back

File: pom.xml (hunk 18 above)
Comment (h-zhi) [reply pom.xml]: pom space indentation has been processed

File: pom.xml (hunk 17 above)
Comment (h-zhi) [reply pom.xml]: pom space indentation has been processed

File: sqlparser/seata-sqlparser-antlr/src/main/java/io/seata/sqlparser/antlr/SQLOperateRecognizerHolderFactory.java (hunk 19 above)
Comment (h-zhi) [reply sqlparser/seata-sqlparser-antlr/src/main/java/io/seata/sqlparser/antlr/SQLOperateRecognizerHolderFactory.java]: already processed

**Review by slievrly — APPROVED** [review]
LGTM

**Review by l81893521 — APPROVED** [review]
LGTM

**Pull request merged**